test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -101,9 +149,342 @@ TEST_P(testNumaNodes, checkNumaNodesAllocations) {
+// allocated on the node of the CPU that triggered the allocation.
+// It will be executed on each available CPU.
+TEST_P(testNumaOnEachCpu, checkModeLocal) {
+    int cpu = GetParam();
+    cpu_set_t *mask = CPU_ALLOC(CPU_SETSIZE);
+    CPU_ZERO(mask);
+
+    CPU_SET(cpu, mask);
+    int ret = sched_setaffinity(0, sizeof(cpu_set_t), mask);
+    UT_ASSERTeq(ret, 0);
+
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+    os_memory_provider_params.numa_mode = UMF_NUMA_MODE_LOCAL;
+    initOsProvider(os_memory_provider_params);
+
+    umf_result_t umf_result;
+    umf_result =
+        umfMemoryProviderAlloc(os_memory_provider, alloc_size, 0, &ptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(ptr, nullptr);
+
+    // Verify we're on the expected CPU
+    int cpu_check = sched_getcpu();
+    UT_ASSERTeq(cpu, cpu_check);
+
+    int numa_node_number = numa_node_of_cpu(cpu);
+    printf("Got CPU: %d, got numa node: %d\n", cpu, numa_node_number);
+
+    // 'ptr' must point to an initialized value before retrieving its numa node
+    memset(ptr, 0xFF, alloc_size);
+    int retrieved_numa_node_number = getNumaNodeByPtr(ptr);
+    ASSERT_EQ(retrieved_numa_node_number, numa_node_number);
+    CPU_FREE(mask);
+}
+
+// Test for allocation on numa node with default mode enabled.
+// Since no policy is set by the set_mempolicy function, it should
+// default to the system-wide default policy, which allocates pages
+// on the node of the CPU that triggers the allocation.
+TEST_F(testNuma, checkModeDefault) {
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+    initOsProvider(os_memory_provider_params);
+
+    umf_result_t umf_result;
+    umf_result =
+        umfMemoryProviderAlloc(os_memory_provider, alloc_size, 0, &ptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(ptr, nullptr);
+
+    int cpu = sched_getcpu();
+    int numa_node_number = numa_node_of_cpu(cpu);
+    printf("Got CPU: %d, got numa node: %d\n", cpu, numa_node_number);
+
+    // 'ptr' must point to an initialized value before retrieving its numa node
+    memset(ptr, 0xFF, alloc_size);
+    int retrieved_numa_node_number = getNumaNodeByPtr(ptr);
+    ASSERT_EQ(retrieved_numa_node_number, numa_node_number);
+}
+
+// Test for allocations on numa nodes with interleave mode enabled.
+// The page allocations are interleaved across the set of nodes specified in nodemask.
+TEST_F(testNuma, checkModeInterleave) {
+    constexpr int pages_num = 1024;
+    size_t page_size = sysconf(_SC_PAGE_SIZE);
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+
+    std::vector<unsigned> numa_nodes = get_available_numa_nodes();
+    set_all_available_nodemask_bits(nodemask);
+
+    os_memory_provider_params.numa_list = &numa_nodes[0];
```

Comment:
nit: `numa_nodes.data()` does the same trick.

Reply:
Done.